presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -469,4 +473,28 @@ public String toString()
                 .add("estimatedSize", estimatedSize)
                 .toString();
     }
+
+    public Iterator<Page> getPages()
+    {
+        return new Iterator<Page>()
+        {
+            private int pageCounter;
+
+            @Override
+            public boolean hasNext()
+            {
+                return pageCounter < channels[0].size();
+            }
+
+            @Override
+            public Page next()
+            {
+                Block[] blocks = Stream.of(channels)
+                        .map(channel -> channel.get(pageCounter))
+                        .toArray(Block[]::new);
+                pageCounter++;
+                return new Page(blocks);
```

Comment:
Also quoting your own comment so you get an email with it and not forget to make changes to this class:

>This is almost correct implementation of Iterator except that next() should throw NoSuchElementException when iterator is exhausted, rather than ArrayIndexOutOfBound or something like that.